joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/parser/RubyNodeCreator.scala

```diff
@@ -358,52 +340,107 @@ class RubyNodeCreator extends RubyParserBaseVisitor[RubyNode] {
       case Nil          => List.empty
```

Comment:
The top case is no longer applicable (assuming it is impossible in the grammar), so it should be dead code. Not affecting functionality, so up to you what to do with it.

Reply:
Good point, I should remove it as it is probably the least eloquent code and may be adding complexity